project/ParadoxSignatureDirective.scala

```diff
@@ -13,18 +13,54 @@ import org.pegdown.ast.{DirectiveNode, HtmlBlockNode, VerbatimNode, Visitor}
 import scala.collection.JavaConverters._
 import scala.io.{Codec, Source}
 
+import _root_.io.github.lukehutch.fastclasspathscanner.FastClasspathScanner
+import _root_.io.github.lukehutch.fastclasspathscanner.scanner.ScanResult
+
+
 object ParadoxSupport {
   val paradoxWithSignatureDirective = Seq(
-    paradoxDirectives += Def.taskDyn {
+    paradoxDirectives ++= Def.taskDyn {
       val log = streams.value.log
-      Def.task {
+      val classpath = (fullClasspath in Compile).value.files.map(_.toURI.toURL).toArray
+      val classloader = new java.net.URLClassLoader(classpath, this.getClass().getClassLoader())
+      val scanner = new FastClasspathScanner().addClassLoader(classloader).scan()
+      val directives = paradoxDirectives.value
+      Def.task { Seq(
         { context: Writer.Context ⇒
-          new SignatureDirective(context.location.tree.label, context.properties, msg ⇒ log.warn(msg))
-        }
-      }
+            new SignatureDirective(context.location.tree.label, context.properties, msg ⇒ log.warn(msg))
+        },
+        { context: Writer.Context ⇒ {
+            val scaladocDirective = directives.map(_.apply(context)).collect { case x: ScaladocDirective => x }.head
+            val javadocDirective = directives.map(_.apply(context)).collect { case x: JavadocDirective => x }.head
+            new UnidocDirective(scaladocDirective, javadocDirective, scanner)
+          }
+        },
+      )}
     }.value
   )
 
+  class UnidocDirective(scaladocDirective: ScaladocDirective, javadocDirective: JavadocDirective, scanner: ScanResult) extends InlineDirective("unidoc") {
+    def render(node: DirectiveNode, visitor: Visitor, printer: Printer): Unit = {
+      val matches = scanner.getNamesOfAllStandardClasses.asScala.filter(_.startsWith("akka.http")).filter(_.endsWith('.' + node.label))
+      if (matches.size > 2)
```

Comment:
I guess that's because we expect one `scaladsl` and one `javadsl` instance?

Reply:
Yes - so indeed this will break in the case where there's 2 matches that are not the `javadsl`/`scaladsl` instances, but that seemed like it might be rare.

Reply:
I guess it should also check that it matches *something*.

Reply:
The next step is matches.foreach, so that's implicit ;). When we figure out how to do this in a way that performs well and does not leak memory we'll also lay it out in an easier to read way :).

Reply:
But `foreach` doesn't fail, right? I think we should at least warn if no match could be found.

Reply:
Aah yes indeed it should!